crates/biome_js_analyze/src/lint/suspicious/no_empty_block_statements.rs

```diff
@@ -109,3 +111,29 @@ fn is_empty(query: &Query) -> bool {
         JsSwitchStatement(statement) => statement.cases().len() == 0,
     }
 }
+
+/// Check if the function is a constructor with TypeScript parameter properties.
+///
+/// https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties
+fn is_constructor_with_ts_param_props(query: &Query) -> bool {
+    use Query::*;
+    let JsFunctionBody(body) = query else {
+        return false;
+    };
+
+    let Some(constructor) = body
+        .syntax()
+        .ancestors()
+        .find_map(|node| node.cast::<JsConstructorClassMember>())
```

Comment:
Would this result in a false negative in the following case?
```ts
class Foo {
    constructor(
        private a: string
    ){
        function bar(){ }
                      ^^^
        bar()
   }
 }
```

Reply:
Yes, it does. Good catch.